react/features/stream-effects/rnnoise/RnnoiseProcessor.js

```diff
@@ -175,23 +133,33 @@ export default class RnnoiseProcessor {
      * Calculate the Voice Activity Detection for a raw Float32 PCM sample Array.
      * The size of the array must be of exactly 480 samples, this constraint comes from the rnnoise library.
      *
-     * @param {Float32Array} pcmFrame - Array containing 32 bit PCM samples.
+     * @param {Float32Array} pcmFrame - Array containing 32 bit PCM samples. Parameter is also used as output
+     * when {@code shouldDenoise} is true.
+     * @param {boolean} shouldDenoise - Should the denoised frame be returned in pcmFrame.
      * @returns {Float} Contains VAD score in the interval 0 - 1 i.e. 0.90 .
      */
-    calculateAudioFrameVAD(pcmFrame: Float32Array) {
-        if (this._destroyed) {
-            throw new Error('RnnoiseProcessor instance is destroyed, please create another one!');
+    calculateAudioFrameVAD(pcmFrame: Float32Array, shouldDenoise: Boolean = false) {
+        // Convert 32 bit Float PCM samples to 16 bit Float PCM samples as that's what rnnoise accepts as input
+        for (let i = 0; i < 480; i++) {
```

Comment:
Pl add a constant for the samples amount with some docs.

Reply:
added